httpobs/scanner/analyzer/headers.py

```diff
@@ -38,76 +82,86 @@ def content_security_policy(reqs: dict, expectation='csp-implemented-with-no-uns
-                   (directive[1].split() if len(directive) > 1 else []) for directive in csp}
-        except:
-            output['result'] = 'csp-header-invalid'
-            return output
-
-        # Replicate default-src to object-src, script-src, and style-src, if they don't exist and default-src does
-        csp['default-src'] = csp.get('default-src', [])
-        for directive in ('object-src', 'script-src', 'style-src'):
-            csp[directive] = csp.get(directive, csp['default-src'])
-
-        # Remove 'unsafe-inline' if nonce or hash are used are in script-src
-        # See: https://github.com/mozilla/http-observatory/issues/88
-        if any(source.startswith(('\'sha256-', '\'sha384-', '\'sha512-', '\'nonce-'))
-               for source in csp.get('script-src', ())):
-            csp['script-src'] = [source for source in csp['script-src'] if source != '\'unsafe-inline\'']
-
-        # Now to make the piggies squeal
-
-        # No 'unsafe-inline' or data: in script-src
-        # Also don't allow overly broad schemes such as https: in either object-src or script-src
-        # Likewise, if you don't have object-src or script-src defined, then all sources are allowed
-        if (any(source in csp['script-src'] for source in dangerously_broad + unsafe_inline)
-                or any(source in csp['object-src'] for source in dangerously_broad)
-                or not csp['object-src']
-                or not csp['script-src']):
-            output['result'] = 'csp-implemented-with-unsafe-inline'
-
-        # If the site is https, it shouldn't allow any http: as a source (passive or mixed content)
-        elif urlparse(response.url).scheme == 'https' and 'http:' in header:
-            output['result'] = 'csp-implemented-with-insecure-scheme'
-        elif '\'unsafe-eval\'' in csp['script-src'] + csp['style-src']:
-            output['result'] = 'csp-implemented-with-unsafe-eval'
-
-        # Don't allow 'unsafe-inline', data:, or overly broad sources in style-src
-        elif any(source in csp['style-src'] for source in dangerously_broad + unsafe_inline):
-            output['result'] = 'csp-implemented-with-unsafe-inline-in-style-src-only'
-
-        # Only if default-src is 'none' and 'none' alone, since additional uris override 'none'
-        elif csp.get('default-src') == ['\'none\'']:
-            output['result'] = 'csp-implemented-with-no-unsafe-default-src-none'
-        else:
-            output['result'] = 'csp-implemented-with-no-unsafe'
+    # First we need to combine the HTTP header and HTTP Equiv "header"
+    try:
+        headers = {
+            'http': __parse_csp(response.headers.get('Content-Security-Policy'))
+            if 'Content-Security-Policy' in response.headers else None,
+            'equiv': __parse_csp(response.http_equiv.get('Content-Security-Policy'))
+            if 'Content-Security-Policy' in response.http_equiv else None,
+        }
+    except:
```

Comment:
Naked `except`s are really bad practice. I know it's catching some expected exceptions here, but it could potentially mask other errors that are unexpected. You should whitelist a set of expected exception types, e.g. `except (ValueError) as e:`

Reply:
My problem is I'm not entirely sure what it might catch here, since the data coming from the client can be so weird. It *should* be just ValueError looking at my code, but I'm not sure if there are some weird subsets of Unicode or null characters or other weirdness that might trigger an exception. I really just want to catch any weird/broken header and mark it as invalid.